### PR TITLE
104262 csi driver for azure keyvault unit tests

### PR DESCRIPTION
**Pull request by satr**

### CODE REVIEW

File: pkg/apis/utils/azurekeyvault.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package utils
+
+import (
+	"github.com/equinor/radix-operator/pkg/apis/kube"
+	v1 "github.com/equinor/radix-operator/pkg/apis/radix/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+//GetRadixAzureKeyVaultObjectTypePtr Gets pointer to RadixAzureKeyVaultObjectType
+func GetRadixAzureKeyVaultObjectTypePtr(objectType v1.RadixAzureKeyVaultObjectType) *v1.RadixAzureKeyVaultObjectType {
```
Comment (nilsgstrabo) [inline pkg/apis/utils/azurekeyvault.go:10]: Function is only used by tests. Can it be moved to test/utils.go instead?

File: pkg/apis/utils/azurekeyvault.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	return &objectType
+}
+
+//GetAzureKeyVaultTypeSecrets Gets secrets with kube.RadixSecretRefTypeLabel and value v1.RadixSecretRefTypeAzureKeyVault
+func GetAzureKeyVaultTypeSecrets(secrets *corev1.SecretList) *corev1.SecretList {
```
Comment (nilsgstrabo) [inline pkg/apis/utils/azurekeyvault.go:15]: Function is only used by tests. Can it be moved to test/utils.go instead?

**Review by nilsgstrabo — APPROVED** [review]
Approved. Just a few minor things you can change if you want to.

File: pkg/apis/deployment/secrets.go [hunk 3]
```diff
@@ -489,7 +333,7 @@ func (deploy *Deployment) listSecrets(labelSelector string) ([]*v1.Secret, error
 func (deploy *Deployment) createOrUpdateSecret(ns, app, component, secretName string, isExternalAlias bool) error {
 	secretType := v1.SecretType(kube.SecretTypeOpaque)
 	if isExternalAlias {
-		secretType = v1.SecretType("kubernetes.io/tls")
+		secretType = "kubernetes.io/tls"
```
[the remark this answers is not in the document]
Comment (satr) [reply pkg/apis/deployment/secrets.go:336]: Used now `k8s.io/api/core/v1/SecretType`

File: pkg/apis/utils/azurekeyvault.go (hunk 1 above)
Comment (satr) [reply pkg/apis/utils/azurekeyvault.go:10]: Moved

File: pkg/apis/utils/azurekeyvault.go (hunk 2 above)
Comment (satr) [reply pkg/apis/utils/azurekeyvault.go:15]: Moved

**Review by nilsgstrabo — DISMISSED** [review]

**Pull request merged**